File: rest-service/src/main/java/mb/api/model/IOShortJobResponse.java
package mb.api.model;

import com.fasterxml.jackson.annotation.JsonGetter;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import mb.api.model.io.JsonKeys;
import mb.lib.model.JobStatus;

@JsonDeserialize(as = IOShortJobResponseImpl.class)
public interface IOShortJobResponse
{
  @JsonProperty(JsonKeys.ID)
  String getId();

  @JsonProperty(JsonKeys.ID)
  IOShortJobResponse setId(String id);

  @JsonProperty(JsonKeys.Description)
  String getDescription();

  @JsonProperty(JsonKeys.Description)
  IOShortJobResponse setDescription(String description);

  @JsonProperty(JsonKeys.Status)
  JobStatus getStatus();

  @JsonProperty(JsonKeys.Status)
  IOShortJobResponse setStatus(JobStatus status);

  @JsonProperty(JsonKeys.Created)
  String getCreated();

  @JsonProperty(JsonKeys.Created)
  IOShortJobResponse setCreated(String date);

  @JsonProperty(JsonKeys.Expires)
  String getExpires();

  @JsonSetter(JsonKeys.Expires)
  IOShortJobResponse setExpires(String date);

  @JsonProperty(JsonKeys.MaxResultSize)
  Long getMaxResultSize();

  @JsonSetter(JsonKeys.MaxResultSize)
  IOShortJobResponse setMaxResultSize(Long val);

  @JsonProperty(JsonKeys.ParentJobs)
  IOParentJobLink[] getParentJobs();

  @JsonSetter(JsonKeys.ParentJobs)
  IOShortJobResponse setParentJobs(IOParentJobLink[] parentJobID);

  @JsonProperty(JsonKeys.IsPrimary)
  boolean getIsPrimary();

  @JsonSetter(JsonKeys.IsPrimary)
  IOShortJobResponse setIsPrimary(boolean isPrimary);

  @JsonProperty(JsonKeys.Site)
  String getSite();

  @JsonSetter(JsonKeys.Site)
  IOShortJobResponse setSite(String site);

  @JsonProperty(JsonKeys.Targets)
  IOJobTarget[] getTargets();

  @JsonSetter(JsonKeys.Targets)
  IOShortJobResponse setTargets(IOJobTarget[] targets);

  @JsonGetter(JsonKeys.IsCached)
  boolean isCached();

  @JsonSetter(JsonKeys.IsCached)
  IOShortJobResponse setIsCached(boolean value);
}

File: rest-service/src/main/kotlin/mb/api/model/IOShortJobResponse.kt
package mb.api.model

import com.fasterxml.jackson.annotation.JsonGetter
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL
import mb.api.model.io.JsonKeys
import mb.lib.model.JobStatus

@JsonInclude(NON_NULL)
open class IOShortJobResponse(
  @get:JsonGetter(JsonKeys.ID)
  val id: String,

  @get:JsonGetter(JsonKeys.Description)
  val description: String?,

  @get:JsonGetter(JsonKeys.Status)
  val status: JobStatus,

  @get:JsonGetter(JsonKeys.Created)
  val created: String?,

  @get:JsonGetter(JsonKeys.MaxResultSize)
  val maxResultSize: Long,

  @get:JsonGetter(JsonKeys.ParentJobs)
  val parentJobs: Array<IOJobLink>?,

  @get:JsonGetter(JsonKeys.ChildJobs)
  val childJobs: Array<IOJobLink>?,

  @get:JsonGetter(JsonKeys.IsPrimary)
  val isPrimary: Boolean,

  @get:JsonGetter(JsonKeys.Site)
  val site: String,

  @get:JsonGetter(JsonKeys.Targets)
  val targets: Array<IOJobTarget>?,

  @get:JsonGetter(JsonKeys.IsRerunnable)
  val isRerunnable: Boolean,
)
